8-web-components/7-shadow-dom-events/article.md

```diff
@@ -167,26 +167,26 @@ inner.dispatchEvent(new CustomEvent('test', {
 </script>
 ```
 
-## Summary
+## 总结
 
-Events only cross shadow DOM boundaries if their `composed` flag is set to `true`.
+事件仅仅是在它们的 `composed` 标志设置为 `true` 的时候才能通过 shadow DOM 边界。
 
-Built-in events mostly have `composed: true`, as described in the relevant specifications:
+内建事件大部分都是 `composed: true` 的，正如相关规范所描述的那样：
 
-- UI Events <https://www.w3.org/TR/uievents>.
-- Touch Events <https://w3c.github.io/touch-events>.
-- Pointer Events <https://www.w3.org/TR/pointerevents>.
-- ...And so on.
+- UI 事件 <https://www.w3.org/TR/uievents>。
+- Touch 事件 <https://w3c.github.io/touch-events>。
+- Pointer 事件 <https://www.w3.org/TR/pointerevents>。
+- ……等等。
 
-Some built-in events that have `composed: false`:
+也有些内建事件它们是 `composed: false` 的：
 
-- `mouseenter`, `mouseleave` (also do not bubble),
-- `load`, `unload`, `abort`, `error`,
-- `select`,
-- `slotchange`.
+- `mouseenter`，`mouseleave`（也不冒泡），
+- `load`，`unload`，`abort`，`error`，
+- `select`，
+- `slotchange`。
 
-These events can be caught only on elements within the same DOM.
+这些事件仅能在同一 DOM 中的元素上捕获。
 
-If we dispatch a `CustomEvent`, then we should explicitly set `composed: true`.
+如果我们发送一个 `CustomEvent`，那么我们应该显示设置 `composed: true`。
 
-Please note that in case of nested components, composed events bubble through all shadow DOM boundaries. So, if an event is intended only for the immediate enclosing component, we can also dispatch it on the shadow host. Then it's out of the shadow DOM already.
+请注意，如果是嵌套组件，一个 shadow DOM 可能嵌套到另外一个 shadow DOM 中。在这种情况下合成事件冒泡到所有 shadow DOM 边界。因此，如果一个事件仅用于直接封闭组件，我们也可以在 shadow host 上调度它并设置 `composed: false`。这样它就不在组件 shadow DOM 中，也不会冒泡到更高级别的 DOM。
```

Comment:
这段话多翻译了一些内容？

Reply:
`dispatch` 的翻译要统一

Reply:
添加的英文内容是之前的，最新的英文是：
Please note that in case of nested components, one shadow DOM may be nested into another. In that case composed events bubble through all shadow DOM boundaries. So, if an event is intended only for the immediate enclosing component, we can also dispatch it on the shadow host and set composed: false. Then it’s out of the component shadow DOM, but won’t bubble up to higher-level DOM.

Reply:
OK